### PR TITLE
[forcefield] updates AdaptiveBeamForceFieldAndMass with the new matrix assembly API 

### PR DESCRIPTION
**Pull request by EulalieCoevoet**
Implements  buildMassMatrix() and buildStiffnessMatrix() for AdaptiveBeamForceFieldAndMass.

### CODE REVIEW

File: src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl [hunk 1]
```diff
@@ -362,6 +364,30 @@ void AdaptiveBeamForceFieldAndMass<DataTypes>::addMToMatrix(const MechanicalPara
...
+        BeamLocalMatrices &bLM = m_localBeamMatrices[b];
+        l_interpolation->getNodeIndices( b,  node0Idx, node1Idx );
+
+        /// matrices in global frame
+        const Matrix6x6 M00 = bLM.m_A0Ref.multTranspose((bLM.m_M00 * bLM.m_A0Ref));
+        const Matrix6x6 M01 = bLM.m_A0Ref.multTranspose((bLM.m_M01 * bLM.m_A1Ref));
+        const Matrix6x6 M10 = bLM.m_A1Ref.multTranspose((bLM.m_M10 * bLM.m_A0Ref));
+        const Matrix6x6 M11 = bLM.m_A1Ref.multTranspose((bLM.m_M11 * bLM.m_A1Ref));
+
+        matrices->add(node0Idx * 6, node0Idx * 6,  M00);
```
Comment (fredroy) [inline src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:384]: There is no Mat6x6 version of add in https://github.com/sofa-framework/sofa/blob/master/Sofa/framework/Core/src/sofa/core/MatrixAccumulator.h
Maybe did you add it in your sofa local branch and would need to PR it in https://github.com/sofa-framework/sofa ? @EulalieCoevoet

Comment (EulalieCoevoet) [reply src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:384]: Indeed !

File: src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl [hunk 2]
```diff
@@ -362,6 +364,30 @@ void AdaptiveBeamForceFieldAndMass<DataTypes>::addMToMatrix(const MechanicalPara
...
+    {
+        unsigned int node0Idx, node1Idx;
+        BeamLocalMatrices &bLM = m_localBeamMatrices[b];
+        l_interpolation->getNodeIndices( b,  node0Idx, node1Idx );
+
+        /// matrices in global frame
+        const Matrix6x6 M00 = bLM.m_A0Ref.multTranspose((bLM.m_M00 * bLM.m_A0Ref));
+        const Matrix6x6 M01 = bLM.m_A0Ref.multTranspose((bLM.m_M01 * bLM.m_A1Ref));
+        const Matrix6x6 M10 = bLM.m_A1Ref.multTranspose((bLM.m_M10 * bLM.m_A0Ref));
+        const Matrix6x6 M11 = bLM.m_A1Ref.multTranspose((bLM.m_M11 * bLM.m_A1Ref));
```
Comment (alxbilger) [inline src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:382]:
```suggestion
        const Matrix6x6 M00 = bLM.m_A0Ref.multTranspose(bLM.m_M00 * bLM.m_A0Ref);
        const Matrix6x6 M01 = bLM.m_A0Ref.multTranspose(bLM.m_M01 * bLM.m_A1Ref);
        const Matrix6x6 M10 = bLM.m_A1Ref.multTranspose(bLM.m_M10 * bLM.m_A0Ref);
        const Matrix6x6 M11 = bLM.m_A1Ref.multTranspose(bLM.m_M11 * bLM.m_A1Ref);

```

File: src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl [hunk 3]
```diff
@@ -685,6 +717,35 @@ void AdaptiveBeamForceFieldAndMass<DataTypes>::addKToMatrix(const MechanicalPara
...
+        l_interpolation->getNodeIndices( b,  node0Idx, node1Idx );
+
+        if(node0Idx==node1Idx)
+            continue;
+
+        // matrices in global frame
+        Matrix6x6 K00 = beamLocalMatrix.m_A0Ref.multTranspose((beamLocalMatrix.m_K00 * beamLocalMatrix.m_A0Ref));
+        Matrix6x6 K01 = beamLocalMatrix.m_A0Ref.multTranspose((beamLocalMatrix.m_K01 * beamLocalMatrix.m_A1Ref));
+        Matrix6x6 K10 = beamLocalMatrix.m_A1Ref.multTranspose((beamLocalMatrix.m_K10 * beamLocalMatrix.m_A0Ref));
+        Matrix6x6 K11 = beamLocalMatrix.m_A1Ref.multTranspose((beamLocalMatrix.m_K11 * beamLocalMatrix.m_A1Ref));
```
Comment (alxbilger) [inline src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:741]:
```suggestion
        Matrix6x6 K00 = beamLocalMatrix.m_A0Ref.multTranspose(beamLocalMatrix.m_K00 * beamLocalMatrix.m_A0Ref);
        Matrix6x6 K01 = beamLocalMatrix.m_A0Ref.multTranspose(beamLocalMatrix.m_K01 * beamLocalMatrix.m_A1Ref);
        Matrix6x6 K10 = beamLocalMatrix.m_A1Ref.multTranspose(beamLocalMatrix.m_K10 * beamLocalMatrix.m_A0Ref);
        Matrix6x6 K11 = beamLocalMatrix.m_A1Ref.multTranspose(beamLocalMatrix.m_K11 * beamLocalMatrix.m_A1Ref);

```

File: src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl [hunk 4]
```diff
@@ -685,6 +717,35 @@ void AdaptiveBeamForceFieldAndMass<DataTypes>::addKToMatrix(const MechanicalPara
     }
 }
 
+template<class DataTypes>
+void AdaptiveBeamForceFieldAndMass<DataTypes>::buildStiffnessMatrix(core::behavior::StiffnessMatrix* matrix)
+{
+    unsigned int numBeams = l_interpolation->getNumBeams();
```
Comment (alxbilger) [inline src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:723]:
```suggestion
    const unsigned int numBeams = l_interpolation->getNumBeams();

```

File: src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl [hunk 5]
```diff
@@ -362,6 +364,30 @@ void AdaptiveBeamForceFieldAndMass<DataTypes>::addMToMatrix(const MechanicalPara
     }
 }
 
+template<class DataTypes>
+void AdaptiveBeamForceFieldAndMass<DataTypes>::buildMassMatrix(sofa::core::behavior::MassMatrixAccumulator* matrices)
+{
+    unsigned int numBeams = l_interpolation->getNumBeams();
```
Comment (alxbilger) [inline src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:370]:
```suggestion
    const unsigned int numBeams = l_interpolation->getNumBeams();

```

**Review by alxbilger — CHANGES_REQUESTED** [review]
Cosmetic

File: src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl [hunk 6]
```diff
@@ -32,6 +32,8 @@
 //
 #pragma once
 
+#include "sofa/core/behavior/BaseLocalForceFieldMatrix.h"
+#include "sofa/core/behavior/BaseLocalMassMatrix.h"
```
Comment (alxbilger) [inline src/BeamAdapter/component/forcefield/AdaptiveBeamForceFieldAndMass.inl:36]:
```suggestion
#include <sofa/core/behavior/BaseLocalForceFieldMatrix.h>
#include <sofa/core/behavior/BaseLocalMassMatrix.h>

```

**Pull request merged**